src/WinRT.Runtime/TypeNameSupport.cs

```diff
@@ -25,12 +26,34 @@ enum TypeNameGenerationFlags
     static class TypeNameSupport
     {
         private static List<Assembly> projectionAssemblies = new List<Assembly>();
+        private static ConcurrentDictionary<string, Type> typeNameCache = new ConcurrentDictionary<string, Type>() { ["TrackerCollection<T>"] = null };
 
         public static void RegisterProjectionAssembly(Assembly assembly)
         {
             projectionAssemblies.Add(assembly);
         }
 
+        /// <summary>
+        /// Parses and loads the given type name, if not found in the cache.
+        /// </summary>
+        /// <param name="runtimeClassName">The runtime class name to attempt to parse.</param>
+        /// <returns>The type, if found.  Null otherwise</returns>
+        public static Type FindTypeByNameCached(string runtimeClassName)
```

Comment:
Should this also be used for https://github.com/microsoft/CsWinRT/blob/ad6adcb997b40cea649cf4156451316e714bc3ea/src/WinRT.Runtime/Projections/Type.cs#L87 or FindTypeByName itself be updated to consult the cache as an internal implementation of the API?

Reply:
I skipped that case, as I think we want the exception to propagate back to native code as an HRESULT